Initial commit of swagger design

floppyears opened the pull request:
CO-439

floppyears committed:
Initial commit of swagger design

CO-439

floppyears committed:
Add grading schemes

CO-439

floppyears committed:
Add instruction-type

CO-439

floppyears committed:
Add links pagination to instruction type

CO-439

floppyears committed:
ERP Organizations

CO-439

floppyears committed:
Course subjects

CO-439

floppyears committed:
Class search

CO-439

magical commented on class-search-swagger.yaml:
```diff
@@ -0,0 +1,275 @@
+          type: string
+          description: "Whether or not the class meets on this day."
+        wednesday:
+          type: string
```
These should be bool. Also, why are all the days out of order?

floppyears replied:
Haha, because I was being a bit lazy yesterday at 6pm, I'll fix it :)

magical commented on term-swagger.yaml:
```diff
@@ -0,0 +1,136 @@
+        description: "Aid year associated with this term"
+      housingStartDate:
+        type: string
+        description: "First day for meal plan and housing assignments"
```
What date format?

magical commented on academic-periods.yaml:
```diff
@@ -0,0 +1,144 @@
+      tags:
+       - academic-periods
+      # Describe this verb here. Note: you can use markdown
+      description: "Gets all `academic-level` objects. A combination of terms, years and subterms"
```
s/academic-level/academic-period/

magical commented on class-search-swagger.yaml:
```diff
@@ -0,0 +1,275 @@
+          schema:
+            $ref: "#/definitions/Errors"
+      security:
+        - locations_api_key: []
```
Change to courses_api_key or delete?

floppyears replied:
Good point. I used to specify to import the swagger and security information into the API gateway, but later on found out that the gateway didn't support this part of the swagger spec :(

magical commented on class-search-swagger.yaml:
```diff
@@ -0,0 +1,275 @@
+
+securityDefinitions:
+
+  locations_api_key:
```
Change to courses_api_key or delete?

magical commented on class-search-swagger.yaml:
```diff
@@ -0,0 +1,275 @@
+        "500":
+          description: "Internal Server Error"
+          schema:
+            $ref: "#/definitions/Errors"
```
Do you really want internal server error to return a list of errors?

floppyears replied:
Good catch!

floppyears committed:
The decision for now is to KISS

CO-439

Removing extra look up tables since they are not directly related
to searching for classes / catalog.

floppyears committed:
Update term swagger

CO-439

floppyears committed:
Update doc on term + descriptions

CO-439

floppyears committed:
Add /term/{code}

CO-439

floppyears committed:
KISS

CO-439

floppyears committed:
Add more descriptions to course subjects

CO-439

floppyears committed:
Simplifying class search

CO-439

Adding more sample data and explanation to class search swagger.

floppyears committed:
Simplifying server errors to use a single error object

CO-439

magical commented on class-search-swagger.yaml:
```diff
@@ -124,48 +113,44 @@ definitions:
       status: 
         type: string
-        description: "Whether or not the course is open, closed"
+        description: "Whether or not the course is open or closed"
```
s/Whether or not/Whether/

magical commented on class-search-swagger.yaml:
```diff
@@ -124,48 +113,44 @@ definitions:
       subject:
         type: string
-        description: "Letter indification to specify the subject"
+        description: "Letter indification to specify the subject. This is the same value as the subject parameter"
```
s/indification/identifier/ ?

magical commented on class-search-swagger.yaml:
```diff
@@ -0,0 +1,250 @@
+        type: string
+        description: "Title of the course section"
+      creditHourHigh:
+        type: string
```
type: integer?

Same for creditHourLow, creditHours, enrollment, maximumEnrollment, waitCapacity, waitCount.

floppyears committed:
Remove useless comment

CO-439

floppyears committed:
Minor fixes based on code review

CO-439
